docker/docker-entrypoint.sh

```diff
@@ -1,47 +1,172 @@
 #!/bin/bash
 set -eo pipefail
 
-# mysql_log logs messages with a timestamp and optional color formatting.
+# mysql_log prints a structured log line with optional color formatting
+#
 # Arguments:
-#   $1 - Log type (e.g., Note, Warn, ERROR)
-#   $@ - Log message (if empty, reads from stdin)
+#   $1            - Timestamp (empty string "" for current time)
+#   $2..$n-1      - Metadata fields (last argument before optional type index is the message)
+#   $n (optional) - Numeric index of metadata field to use for determining log type/color
```

Comment:
What does this mean? This was a lot more clear in the previous version ("Log Type (e.g. Note, Warn, Error)"). The new description is pretty opaque and much harder to call. Logging frameworks generally specify the log level as the first argument, so it feels a bit strange to see it as an optional argument that is an opaque index into a "metadata field". Cleaning this function interface up a bit will make this a lot easier for future maintainers to work with.